object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionChecker.java

```diff
@@ -79,18 +85,10 @@
     return result;
   }
 
-  /**
-   * Adds special reporting for method.invocation.invalid errors to turn them into
-   * finalizer.invocation.invalid errors.
-   */
   @Override
-  public void report(final Result r, final Object src) {
-
-    Result theResult = r;
-
-    String errKey = r.getMessageKeys().iterator().next();
-    if ("method.invocation.invalid".equals(errKey) && r.isFailure()) {
-      Object[] args = r.getDiagMessages().iterator().next().getArgs();
+  public void reportError(Object source, @CompilerMessageKey String messageKey, Object... args) {
+    String errKey = messageKey;
```

Comment:
@kelloggm was there some other way you wanted to implement the custom error message, based on some new Checker Framework method?

Reply:
Yes, but I can make that change separately after this lands if that's easier for you all.

Reply:
That's fine, as long as we don't forget